src/js/AccessibilityUtils.js

```diff
@@ -1041,30 +1041,99 @@ axs.utils.getQuerySelectorText = function(obj) {
- * ID reference list or single ID reference.
- * @param {!string} attributeName Name of an ARIA attribute, e.g. 'aria-owns'.
+ * Gets elements that refer to this element in an ARIA attribute that takes an ID reference list or
+ * single ID reference.
  * @param {Element} element a potential referent.
+ * @param {string=} opt_attributeName Name of an ARIA attribute to limit the results to, e.g. 'aria-owns'.
  * @return {NodeList} The elements that refer to this element.
  */
-axs.utils.getIdReferrers = function(attributeName, element) {
+axs.utils.getAriaIdReferrers = function(element, opt_attributeName) {
+    var propertyToSelector = function(propertyKey) {
+        var propertyDetails = axs.constants.ARIA_PROPERTIES[propertyKey];
+        if (propertyDetails) {
+            if (propertyDetails.valueType === ('idref')) {
+                return '[aria-' + propertyKey + '=\'' + id + '\']';
+            } else if (propertyDetails.valueType === ('idref_list')) {
+                return '[aria-' + propertyKey + '~=\'' + id + '\']';
+            }
+        }
+        return '';
+    };
     if (!element)
         return null;
     var id = element.id;
-    var propertyKey = attributeName.replace(/^aria-/, '');
-    var property = axs.constants.ARIA_PROPERTIES[propertyKey];
-    if (!id || !property)
+    if (!id)
         return null;
-    var propertyType = property.valueType;
-    if (propertyType === 'idref_list' || propertyType === 'idref') {
-        id = id.replace(/'/g, "\\'");
-        var referrerQuery = "[" + attributeName + "~='" + id + "']";
-        return element.ownerDocument.querySelectorAll(referrerQuery);
+    id = id.replace(/'/g, "\\'");  // make it safe to use in a selector
+
+    if (opt_attributeName) {
+        var propertyKey = opt_attributeName.replace(/^aria-/, '');
+        var referrerQuery = propertyToSelector(propertyKey);
+        if (referrerQuery) {
+            return element.ownerDocument.querySelectorAll(referrerQuery);
+        }
+    } else {
+        var selectors = [];
+        for (var propertyKey in axs.constants.ARIA_PROPERTIES) {
+            var referrerQuery = propertyToSelector(propertyKey);
+            if (referrerQuery) {
+                selectors.push(referrerQuery);
+            }
+        }
+        return element.ownerDocument.querySelectorAll(selectors.join(','));
     }
     return null;
 };
 
 /**
+ * Gets elements that refer to this element in an HTML attribute that takes an ID reference list or
+ * single ID reference.
+ * @param {Element} element a potential referent.
+ * @return {NodeList} The elements that refer to this element.
+ */
+axs.utils.getHtmlIdReferrers = function(element) {
+    if (!element)
+        return null;
+    var id = element.id;
+    if (!id)
+        return null;
+    id = id.replace(/'/g, "\\'");  // make it safe to use in a selector
+    var selectorTemplates = ['[contextmenu=\'{id}\']', '[itemref~=\'{id}\']', 'button[form=\'{id}\']',
```

Comment:
Sorry, got distracted halfway through reviewing this! Could you make these one per line? I realise it'll be a lot of lines (20 or so?), but it's pretty hard to read like this (plus it'll make it easier to add any new ones if we realise something's missing).